app/cluster/state/etcd.go

```diff
@@ -297,18 +297,14 @@ func (manager *ETCDManager) WorkspaceIDs(ctx context.Context) <-chan workspace.C
 			for _, event := range watchResp.Events {
 				switch event.Type {
 				case mvccpb.PUT:
-					select {
-					case resultChan <- manager.unmarshalWorkspace(event.Kv.Value):
-					case <-ctx.Done():
-					}
+					resultChan <- manager.unmarshalWorkspace(event.Kv.Value)
```

Comment:
This change would introduce a bug when:
1. the `ctx` gets canceled
2. `resultChan` is either full or its readers get terminated due to the same `ctx` cancelation or other reasons

This would result in an edge case where we would block indefinitely when readers cannot drain the channel and the context gets canceled at the same time (or right before). Thus the context will not get honoured.